src/sentry/spans/consumers/process/strategy.py

```diff
@@ -0,0 +1,64 @@
+import time
+from typing import Any, Generic, TypeVar, Union
+
+from arroyo.processing.strategies.abstract import ProcessingStrategy
+from arroyo.types import Commit, FilteredPayload, Message
+from arroyo.utils.metrics import get_metrics
+
+TPayload = TypeVar("TPayload")
+
+
+class CommitSpanOffsets(ProcessingStrategy[Union[FilteredPayload, TPayload]], Generic[TPayload]):
```

Comment:
Do we have to overwrite the whole class? Can't we inherit from the `CommitOffset` class and just overload `__init__` and `close`?

Reply:
oops yes, I was trying to access some of the private attributes from the parent class at one point, but don't need to do that anymore so I can't just inherit